aws/ec2.go

```diff
@@ -0,0 +1,84 @@
+package aws
+
+import (
+	"fmt"
+
+	awsgo "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/gruntwork-io/aws-nuke/logging"
+)
+
+var ec2Instances = make(map[string][]*string)
+
+// Build string that'll be shown in resources list
+func buildEntryName(instance ec2.Instance) string {
+	return fmt.Sprintf("ec2-%s-%s", *instance.InstanceId, *instance.InstanceType)
+}
+
+// Returns a formatted string of EC2 instance ids
+func getAllEc2Instances(session *session.Session, region string) ([]string, error) {
+	svc := ec2.New(session)
+
+	params := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: awsgo.String("instance-state-name"),
+				Values: []*string{
+					awsgo.String("running"), awsgo.String("pending"),
```

Comment:
Any reason to filter by state? For example, any reason not to nuke `stopped` instances?

Reply:
I filter by state so that I don't get EC2 instances that have already been `terminated` but not yet deleted (i.e. they still appear in the instances list). As for `stopped` instances, my thinking is that since they're aren't incurring any charges there's no need to nuke them. Although now that I think about it, in the spirit of the tool it should just nuke 'em all

Reply:
yea, I'd nuke everything to keep it simple :)

Reply:
I updated this to nuke every state except `terminated` and `shutting-down`. `terminated` is for already terminated but not cleaned up instances, `shutting-down` is for instances that are in the process of being terminated I didn't see a reason to include this two since they represent already terminated or being terminated instances

Reply:
Sounds good!